src/sdk.ts

```diff
@@ -95,11 +91,28 @@ export class BadgerSDK {
       network: this.config.network,
       baseURL: this.api.baseURL,
     });
+    this.graph = new BadgerGraph({
+      network: this.config.network,
+    });
   }
 
   updateProvider(provider: SDKProvider) {
-    this.provider = new providers.MulticallProvider(provider);
-    this.signer = provider.getSigner();
+    const sdkProvider = BadgerSDK.getSdkProvider(provider);
+
+    this.provider = new providers.MulticallProvider(sdkProvider);
+    this.signer = sdkProvider.getSigner();
+  }
+
+  private static getSdkProvider(provider: SDKProvider | string): SDKProvider {
```

Comment:
what is the purpose of having it as static and private?

Reply:
- no link on instance, so it`s static
- didnt know if it would be usefull for users